crates/factor-outbound-pg/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+pub mod client;
+mod host;
+
+use client::Client;
+use spin_factor_outbound_networking::{OutboundAllowedHosts, OutboundNetworkingFactor};
+use spin_factors::{
+    anyhow, ConfigureAppContext, Factor, InstanceBuilders, PrepareContext, RuntimeFactors,
+    SelfInstanceBuilder,
+};
+use tokio_postgres::Client as PgClient;
+
+pub struct OutboundPgFactor<C = PgClient> {
+    _phantom: std::marker::PhantomData<C>,
+}
+
+impl<C: Send + Sync + Client + 'static> Factor for OutboundPgFactor<C> {
+    type RuntimeConfig = ();
+    type AppState = ();
+    type InstanceBuilder = InstanceState<C>;
+
+    fn init<T: RuntimeFactors>(
+        &mut self,
+        mut ctx: spin_factors::InitContext<T, Self>,
+    ) -> anyhow::Result<()> {
+        ctx.link_bindings(spin_world::v1::postgres::add_to_linker)?;
+        ctx.link_bindings(spin_world::v2::postgres::add_to_linker)?;
+        Ok(())
+    }
+
+    fn configure_app<T: RuntimeFactors>(
+        &self,
+        _ctx: ConfigureAppContext<T, Self>,
+    ) -> anyhow::Result<Self::AppState> {
+        Ok(())
+    }
+
+    fn prepare<T: RuntimeFactors>(
+        &self,
+        _ctx: PrepareContext<Self>,
+        builders: &mut InstanceBuilders<T>,
+    ) -> anyhow::Result<Self::InstanceBuilder> {
+        let allowed_hosts = builders
+            .get_mut::<OutboundNetworkingFactor>()?
+            .allowed_hosts();
+        Ok(InstanceState {
+            allowed_hosts,
+            connections: Default::default(),
+        })
+    }
+}
+
+impl<C> Default for OutboundPgFactor<C> {
+    fn default() -> Self {
```

Comment:
Nit: I like having a `new` static function in addition to `Default`. The `new` function would just call `Self::default`. I like it, because it shows up more prominently in the docs and is therefore IMO more discoverable.